[Add] Customer/Product App - models.py 작성

joey-younggap opened the pull request:
## 신규 작성 코드
- Customer/Product App - models.py 작성

## 변경 사항
- urls.py에서 에러 발생요소 수정 : admin url 제거
- MySql 연결할 수 있도록 settings.py 수정 (본인 PC에서 my_settings.py 수정하면 연결됨)

Commit:
[add] Customer / Product models.py 작성
[fix] urls.py admin 에러 수정
[setting] MySql 연결할 수 있도록 settings.py 수정

evergreen-david commented on customer/models.py:
```diff
@@ -1,3 +1,72 @@
 
-# Create your models here.
+class Customer(models.Model):
+    email    = models.CharField(max_length=50)
```
```suggestion
    email    = models.EmailField(max_length=50)
```

evergreen-david commented on customer/models.py:
```diff
@@ -1,3 +1,72 @@
+    password = models.CharField(max_length=300) 
+    name     = models.CharField(max_length=300)
+    phone    = models.CharField(max_length=50)
+    coupons   = models.ManyToManyField('Coupon', related_name='coupons', through='CouponCustomer')
```
```suggestion
    coupons  = models.ManyToManyField('Coupon', related_name='coupons', through='CouponCustomer')
```

evergreen-david commented on customer/models.py:
```diff
@@ -1,3 +1,72 @@
+    account   = models.CharField(max_length=50)
+    name      = models.CharField(max_length=50)
+    phone     = models.CharField(max_length=50)
+    email     = models.CharField(max_length=50)
```
```suggestion
    email     = models.EmailField(max_length=50)
```

evergreen-david commented on customer/models.py:
```diff
@@ -1,3 +1,72 @@
+    card_number     = models.CharField(max_length=18)
+    valid_month     = models.CharField(max_length=2)
+    valid_year      = models.CharField(max_length=2)
+    password        = models.CharField(max_length=2)    
```
카드 비밀번호 2자리만 사용해서 2로 주신건지 확인 부탁드립니다.

joey-younggap replied:
네, 맞습니다. 카드번호 2자리만 사용합니다.

evergreen-david commented on freeporkbelly_main/settings.py:
```diff
@@ -104,12 +105,14 @@
-        'NAME': os.path.join(BASE_DIR, 'db.sqlite3'),
-    }
-}
+#DATABASES = {
+#    'default': {
+#        'ENGINE': 'django.db.backends.sqlite3',
+#        'NAME': os.path.join(BASE_DIR, 'db.sqlite3'),
+#    }
+#}
```
주석은 남기지말고 삭제시켜 주세요.

evergreen-david commented on customer/models.py:
```diff
@@ -1,3 +1,72 @@
+
+class Point(models.Model):
+    customer   = models.ForeignKey(Customer, on_delete=models.SET_NULL, null=True)
+    created_at = models.DateTimeField()
```
created_at 은 최초 한번만 만들어 지기 때문에 auto_now_add = True 속성을 주면 데이터 베이스가 만들어질때 자동으로 입력되기 때문에 편리합니다.

evergreen-david commented on customer/models.py:
```diff
@@ -1,3 +1,72 @@
+    created_at   = models.DateTimeField(auto_now_add=True)
+    name         = models.CharField(max_length=200)
+    condition    = models.CharField(max_length=500)
+    issued_at    = models.DateTimeField() 
```
발급일도 최초 생길때 시간이면 auto_now_add = True 로 지정할수 있지않을까요 ?

evergreen-david commented on customer/models.py:
```diff
@@ -1,3 +1,72 @@
+    customer   = models.ForeignKey(Customer, on_delete=models.SET_NULL, null=True)
+    created_at = models.DateTimeField()
+    comment    = models.CharField(max_length=45)
+    amount     = models.DecimalField(max_digits=10, decimal_places=2)
```
포인트도 소수점 필요할까요?

evergreen-david commented on product/models.py:
```diff
@@ -1,3 +1,82 @@
+    class Meta:
+        db_table = 'option_items'
+
+
+
+
+
+
+
```
공백은 정리해주세요.

evergreen-david commented on requirements.txt:
```diff
@@ -2,8 +2,8 @@ asgiref==3.2.7
 chardet==3.0.4
-conda==4.8.3
-conda-package-handling==1.7.0
+#conda==4.8.3
+#conda-package-handling==1.7.0
```
주석은 정리해주세요

evergreen-david commented on product/models.py:
```diff
@@ -1,3 +1,82 @@
+        db_table = 'products'
+
+class ProductDescription(models.Model):
+    first_img_url = models.URLField(max_length=2000)
+    second_img_url = models.URLField(max_length=2000)
+    third_img_url = models.URLField(max_length=2000)
```
상품별 사진 무조건 3개다 가지는걸까요? 일대다로 정리할 수 있지않을까요 나중에 확장성 고려하더라도 일대다로 모델링하는게 좋을것같습니다. 지금은 3개 고정적이지만 3개이상 가질수 있게 바뀌는 경우에 대해서 확장성 고려하면 좋을 것 같습니다.

joey-younggap replied:
상품별로 사진 2개 가지는 것도 있고, 3개 가지는 것도 있습니다.
그리고 각 사진들은 순서대로 화면에 뿌려져야 합니다.
일대다 구조로 간다고 하면, 각 상품별로 사진을 DB에 순서대로 올려서 나중에 order_by('id')로 하면
상품당 사진이 1번, 2번, 3번 순서대로 들어갈 수 있을까요?

evergreen-david replied:
네 id로 order by 한다면 순서대로 올려야 한다는 가정이 필요한데 낭비되거나 확장성 고려한다면 일대다로 하는게 좋아보입니다.

Commit:
[Fix] Customer / Product Models
- 피드백 사항 반영해서 수정
- 카테고리 클래스 신선일자, 문구 추가 (카테고리별로 동일하게 가기로 결정)
- 프로덕트 클래스 unit_price_comment와 sales_price_comment 추가 : 상품별로단위가 상이함
- 상품 설명사진은 상품과  일대다 관계로 수정

[Fix] settings. requirements.txt
- 피드백 사항 반영해서 수정

evergreen-david commented:
누락된 커멘트 확인 부탁 드립니다.
https://github.com/wecode-bootcamp-korea/FreePorkBelly-backend/pull/1/files#r431583748
